contracts/contracts/ProjectRegistry.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+/**
+ * @title ProjectRegistry
+ * @notice todo
+ * @dev todo
+ */
+contract ProjectRegistry {
+    // Types
+
+    // The pointer to the external metadadata composed by protocol and pointer
+    struct MetaPtr {
+        uint256 protocol;
+        string pointer;
+    }
+
+    // The project structs contains the minimal data we need for a project
+    struct Project {
+        uint96 id;
+        address recipient;
+        MetaPtr metadata;
+    }
+
+    // A linked list of owners of a project
+    // The use of a linked list allows us to easily add and remove owners,
+    // access them directly in O(1), and loop through them.
+    //
+    // {
+    //     count: 3,
+    //     list: {
+    //         OWNERS_LIST_SENTINEL => owner1Address,
+    //         owner1Address => owner2Address,
+    //         owner2Address => owner3Address,
+    //         owner3Address => OWNERS_LIST_SENTINEL
+    //     }
+    // }
+    struct OwnersList {
+        uint256 count;
+        mapping(address => address) list;
+    }
```

Comment:
Oooo how about using openzepplins Role.
Those are solid tested and work well. Have a look at : 

https://github.com/gitcoinco/grants-round/blob/main/packages/contracts/contracts/GrantRoundImplementation.sol#L101

Reply:
@thelostone-mc can they have multiple iterable owners? I thought they had just multiple roles

Reply:
just to be sure I can get the list of owners without knowing them before

Reply:
@thelostone-mc  another thing is that we need owners for Projects and not for the contract itself, so I'm not sure we can do it with that

Reply:
It looks like they had a Roles implementation that does something similar in their 2.X version of contracts. Its pretty similar to the linked list method but uses boolean within the list. https://github.com/hiddentao/openzeppelin-solidity/blob/master/contracts/access/Roles.sol

I gave a shot at something pretty similar here: https://github.com/gitcoinco/grants-hub/pull/56 lmk what you think!

Reply:
yeah I commented in the other PR, that was the initial implementation but we switch to linked list to be sure once you log in you can have the full list of owners that can edit the project with you, otherwise there's no way to do it. the implementation is the same of gnosis safe.

Reply:
ah ok I didn't know this had already been attempted and thought it was worth exploring. It wouldn't be nearly as easy, but wouldn't you be able to derive the owners from emitting events?

Reply:
it would work but I wouldn't rely on events for owners/permissions and in general project details. so we know the most importar page in the project registry doesn't rely on anything else outside dapp+contract. wdyt?

Reply:
That sounds good to me. Happy to go with the linked list

Reply:
> it would work but I wouldn't rely on events for owners/permissions and in general project details. so we know the most importar page in the project registry doesn't rely on anything else outside dapp+contract. wdyt?

One more question here. We are relying on events to display the projects a currently connected user owns, this seems similar in importance to all of the owners of a current project. Just curious what the difference would be in usage and why it would be ok it one instance and not the other?

Reply:
Very true I think for :` If I'm checking the owners of a project for security, and the data is not synced yet, it might be a problem.` it would come down to if you need to verify a list of addresses/owners or just one at a time since there could be an efficient `activeProjectOwner` getter for a single address. Not sure when we would have to check if a group of addresses owns a project but could be possible 🤷 

Just thought I'd bring this up to be consistent with how we are currently fetching projects a user owns